app/controllers/application_controller.rb

```diff
@@ -22,8 +22,8 @@ def current_user_is_admin
         logged_in? && current_user.admin
     end
 
-    def current_user_is_post_author(post)
-      logged_in? && current_user == post.user
+    def current_user_is_author(post_or_comment)
```

Comment:
ja generalnie nie jestem fanem takiego "DRY" na siłe. 

Wiem, że w Railsach jest to podejście żeby robić jak najwięcej rzeczy DRY i się don't repeat yourself, ale z mojego doświadczenia to w długim okresie sprawia że kod jest cięzki do utrzymania. 

Tutaj niech zostanie narazie, ale generalnie ja nie uważam że applicationController powinien być odpowiedzialny za takie sprawdzenie, a bardziej jakiś model. Dodatkowo właśnie nie lubię tej abstrakcji, że i post i comment pozwala sprawdzić czy jest ktoś autorem.

Wolałbym oddzielnie dla postu i oddzielnie dla komenta. Tak, to będzie duplikacja kodu. Ale potem dodasz jakiś inny warunek dla postów i się okaże że twoja abstrakcja ponad postem i komentem przestanie działać i będziesz miał problem jak to napisać

Reply:
Poprawiłem w sumie. Masz rację.